### PR TITLE
7 post comments by article

### PR DESCRIPTION
**Pull request by AmadReihun**
complete task 7

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -178,72 +178,128 @@ describe("app", () => {
...
+  
+          // Checking the accuracy of the data for article_id 1
+  
+          expect(comments[0].comment_id).toBe(5);
+          expect(comments[0].body).toBe('I hate streaming noses');
+          expect(comments[0].article_id).toBe(1);
+          expect(comments[0].author).toBe('icellusedkars');
+          expect(comments[0].votes).toBe(0);
+          expect(comments[0].created_at).toBe('2020-11-03T21:00:00.000Z');
+
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:214]: I would check that **every** comment you have has the article_id of 1

File: __tests__/app.test.js [hunk 2]
```diff
@@ -178,72 +178,128 @@ describe("app", () => {
...
+    });
+    test("returns newly created object", () => {
+      return request(app).post("/api/articles/1/comments").send({
+        username: 'butter_bridge',
+        body: "new comment"
+      })
+        .expect(201)
+        .then((response) => {
+          expect(response.body.comment).toBe("new comment");
+        });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:268]: I would check all the values here and that it has a comment_id to make it more robust

File: __tests__/app.test.js [hunk 3]
```diff
@@ -264,41 +266,49 @@ describe("app", () => {
...
+
+    test("to get error 400 and respond with appropriate message when given invalid and bad type article_id which will get 22P02", () => {
+      return request(app)
+        .get("/api/articles/apple/comments")
+        .expect(400)
+        .then(({body}) => {
+          const message = body.msg;
+          expect(message).toBe("Bad Request")
+        })
+    });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:311]: Good error testing! One more you can add is to check what happens if the object is malformed. So if the username key is missing or the body is not a string for example?

File: __tests__/app.test.js (hunk 1 above)
Comment (AmadReihun) [reply __tests__/app.test.js:214]: Done

File: __tests__/app.test.js (hunk 2 above)
Comment (AmadReihun) [reply __tests__/app.test.js:268]: Thanks done

File: __tests__/app.test.js (hunk 3 above)
Comment (AmadReihun) [reply __tests__/app.test.js:311]: Done

**Pull request merged**